drivers/painter/gc9a01/qp_gc9a01.c

```diff
@@ -141,6 +141,12 @@ painter_device_t qp_gc9a01_make_spi_device(uint16_t panel_width, uint16_t panel_
             driver->spi_dc_reset_config.spi_config.mode            = spi_mode;
             driver->spi_dc_reset_config.dc_pin                     = dc_pin;
             driver->spi_dc_reset_config.reset_pin                  = reset_pin;
+
+            if (!qp_internal_register_device((painter_device_t)driver)) {
+                memset(driver, 0, sizeof(tft_panel_dc_reset_painter_device_t));
+                return NULL;
+            }
```

Comment:
This code could be moved to a function on the panel-type file (`drivers/painter/tft_panel/qp_tft_panel.c`) and call it, to avoid duplication. And that one could also call a common function on QP's internals that receives the pointer + size of struct

Reply:
Could, but won't.
OLED integration, for instance, has a lot more "internal" logic with respect to ownership of surface buffers and the like.
The TFT's might be relatively consistent, but the pattern isn't generic.